frontend/src/components/pages/CourseOverview.tsx

```diff
@@ -1,28 +1,102 @@
 import { ChevronLeftIcon } from "@chakra-ui/icons";
-import { Box, Button, Flex, Text, VStack } from "@chakra-ui/react";
+import {
+  Box,
+  Button,
+  Center,
+  Flex,
+  Image,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
 import { useQuery } from "@apollo/client";
 import Banner from "../learner/Banner";
 import ModuleLessonCount from "../learner/ModuleLessonCount";
 import { GET_COURSE } from "../../APIClients/queries/CourseQueries";
-import { HOME_PAGE } from "../../constants/Routes";
+import {
+  GET_COURSE_PROGRESS,
+  GET_MODULE_PROGRESS,
+} from "../../APIClients/queries/ProgressQueries";
+import { DEFAULT_IMAGE } from "../../constants/DummyData";
+import {
+  COURSE_OVERVIEW_BASE_ROUTE,
+  HOME_PAGE,
+  LOGIN_PAGE,
+} from "../../constants/Routes";
 import { CourseOverviewParams } from "../../types/CourseOverviewTypes";
-import { ModuleType } from "../../types/ModuleEditorTypes";
+import { ModuleProgressType, ModuleType } from "../../types/ModuleEditorTypes";
+import AuthContext from "../../contexts/AuthContext";
 
 const CourseOverview = (): React.ReactElement => {
   const history = useHistory();
   const { courseID }: CourseOverviewParams = useParams();
+  const { authenticatedUser } = useContext(AuthContext);
+
   const { data: courseData } = useQuery(GET_COURSE, {
     variables: {
       id: courseID,
     },
   });
 
+  const { data: courseProgress } = useQuery(GET_COURSE_PROGRESS, {
+    variables: {
+      userId: authenticatedUser?.id,
+      courseIds: [courseID],
+    },
+  });
+
+  const { data: moduleProgress } = useQuery(GET_MODULE_PROGRESS, {
+    variables: {
+      userId: authenticatedUser?.id,
+      courseId: courseID,
+    },
+  });
+
   const onReturnToCourseOverviewClick = () => {
     history.push(HOME_PAGE);
   };
 
+  const onLogInClick = () => {
+    history.push(LOGIN_PAGE);
+  };
+
+  const onCTAClick = () => {
+    let moduleIndex;
+
+    if (
+      !courseProgress?.courseProgress?.length ||
+      courseProgress?.courseProgress?.[0]?.completedAt !== null
+    ) {
+      //  Go to first link
+      moduleIndex = 0;
+    } else {
+      //  Go to first In Progress moduleIndex
+      moduleIndex = moduleProgress?.moduleProgress?.findIndex(
+        (module: ModuleProgressType) =>
+          module?.startedAt !== null && module?.completedAt === null,
+      );
+
+      if (moduleIndex === -1) {
+        //  Go to first Not Started moduleIndex (1st Case: Skip Case)
+        moduleIndex = moduleProgress?.moduleProgress?.findIndex(
+          (module: ModuleProgressType) =>
+            module?.startedAt === null && module?.completedAt === null,
+        );
+
+        if (moduleIndex === -1) {
+          //  Go to first Not Started moduleIndex (2nd Case)
+          moduleIndex = courseData?.course?.modules?.findIndex(
+            (module: ModuleType, index: number) =>
+              index >= moduleProgress?.moduleProgress?.length &&
+              module !== null,
+          );
+        }
+      }
+    }
```

Comment:
At the end of all these if statements, `moduleIndex` may be null still if there was missing data. Could you add a check here and disable the "start" button if it fails?

Reply:
Done